### ISSUE TITLE
SKR 1.4 Turbo vs Tft MKS H43

### ISSUE THREAD

**Issue by andyrok** [bigtreetech/BIGTREETECH-SKR-V1.3#567]
Can someone help me I have a SKR 1.4 Turbo and I bought a Tft MKS H43 because in the specifications they gave it compatible with the SKR .. unfortunately I can't make it work ... it turns on and stays on the logo screen..😱

**Comment by radek8** [comment]
Send config and photo involved

**Comment by andyrok** [comment]
This is the suggest wiring....and the serial port...

<img width="570" alt="Schermata 2022-08-31 alle 16 04 01" src="https://user-images.githubusercontent.com/49729028/189473260-0e123aee-33b5-42dd-bd20-26c3ba2d3f48.png">

But in Configuration_adv_h( marlin for SKR 1.4 Turbo)  "define LCD_SERIAL_PORT" is not present...
<img width="460" alt="Schermata 2022-09-10 alle 09 18 06" src="https://user-images.githubusercontent.com/49729028/189473526-0472a97f-251f-40ba-b75c-f09314b4ac46.png">

MKS wiring suggest ice this

<img width="476" alt="Schermata 2022-09-10 alle 09 12 42" src="https://user-images.githubusercontent.com/49729028/189473382-e9947f41-0703-4cf0-95c5-583e55dc1570.png">


i have try every possibility of wiring and serial port but the result is the same... only logo.....
<img width="524" alt="Schermata 2022-09-10 alle 09 22 02" src="https://user-images.githubusercontent.com/49729028/189473624-649947e7-9f9b-450e-9502-07adc18dac7c.png">

**Comment by radek8** [comment]
Did you enable configuration.h?
#define DGUS_LCD_UI_MKS

Which serial port did you connect the display to? You can't see it from your photos.
In the configuration_adv.h file, add a line with the port number.
#define LCD_SERIAL_PORT

You must not define the same port as a standard serial port in the configuration.h file

**Comment by radek8** [comment]
If you do not define a serial port manually, serial port 3 will be used

![image](https://user-images.githubusercontent.com/46979052/189500708-1405b8e7-8721-4b91-a8c8-09d10354cef6.png)

**Comment by andyrok** [comment]
<img width="291" alt="Schermata 2022-09-10 alle 23 48 50" src="https://user-images.githubusercontent.com/49729028/189502987-756c69cc-f46b-4d43-809f-fd8ffa4745d8.png">
<img width="227" alt="Schermata 2022-09-10 alle 23 50 07" src="https://user-images.githubusercontent.com/49729028/189503001-ae437ea2-528d-4127-9a63-9457cc06e432.png">

This is the port suggest by MKS....
I use Marlin for SKR 1.4 turbo but  the line whit "LCD_SERIAL_PORT" are missed....
ok for add line "LCD_SERIAL_PORT"
I've try but only Logo appears..

**Comment by radek8** [comment]
What port number did you add?

**Comment by radek8** [comment]
Attention WIFI connector has only 3.3V power supply.

**Comment by radek8** [comment]
Your involvement is wrong. The connection of the adapter does not correspond to the connection of the WiFi connector.

**Comment by radek8** [comment]
The adapter will probably be compatible with the AUX1 connector on the SKR V1.3, but you have SKRV 1.4 and the tan does not have an AUX1 connector but WiFi.
These two connectors are not wired the same way.
Be happy if you don't destroy anything.

![image](https://user-images.githubusercontent.com/46979052/189518675-fbd36ed4-558d-42cc-92e0-b08399bb7471.png)

**Comment by radek8** [comment]
![image](https://user-images.githubusercontent.com/46979052/189518751-ac6ad4a7-4a42-44d5-9b34-8530866ad2b1.png)

![image](https://user-images.githubusercontent.com/46979052/189518786-f60e23f1-7882-44a1-ad7b-d1f44c635047.png)

**Comment by andyrok** [comment]
> The adapter will probably be compatible with the AUX1 connector on the SKR V1.3, but you have SKRV 1.4 and the tan does not have an AUX1 connector but WiFi. These two connectors are not wired the same way. Be happy if you don't destroy anything.
> 
> ![image](https://user-images.githubusercontent.com/46979052/189518675-fbd36ed4-558d-42cc-92e0-b08399bb7471.png)

MKS expert give me this information:

“The wifi interface is also a serial port

I thought this was right but some doubts come to me... as for the Tft power supply I think if there were only 3.3 V it wouldn't turn on...

**Comment by andyrok** [comment]
<img width="449" alt="Schermata 2022-09-11 alle 12 17 24" src="https://user-images.githubusercontent.com/49729028/189522413-55a11848-2b15-4d79-a0fc-11e861fde48c.png">

**Comment by radek8** [comment]
Yes, the wifi connector has a serial port, but on other pins. So you can't use a 1:1 cable

**Comment by radek8** [comment]
You still haven't sent a photo of the wiring and files with the current Marlin configuration

**Comment by radek8** [comment]
Make your own connection cable. 
In the evening I will send it a suggestion for his involvement.

**Comment by andyrok** [comment]
<img width="792" alt="Schermata 2022-09-11 alle 17 10 06" src="https://user-images.githubusercontent.com/49729028/189535037-8cc511c5-d133-48ac-885d-fffb26f8a32e.png">
I added the line: "LCD_SERIAL_PORT"

<img width="423" alt="Schermata 2022-09-11 alle 17 11 17" src="https://user-images.githubusercontent.com/49729028/189535101-e633be2d-2cd0-4023-8d3e-59443b0da5ed.png">

<img width="294" alt="Schermata 2022-09-11 alle 17 11 05" src="https://user-images.githubusercontent.com/49729028/189535102-f498bd11-6674-41a0-b685-94a5d8261a50.png">

<img width="406" alt="Schermata 2022-09-11 alle 17 16 28" src="https://user-images.githubusercontent.com/49729028/189535248-f5d699c0-6b15-435f-bfd8-1dd61b84ae00.png">
I created the link with single cables... As in the scheme....

**Comment by radek8** [comment]
If you want to use the WiFi connector, you must define serial port 3
#define LCD_SERIAL_PORT 3

The display requires a 5V power supply. use the 5V pin on the TFT connector.

Did you uncomment this line?
#define DGUS_LCD_UI_MKS

Comment
#define serial_port_2 1

**Comment by radek8** [comment]
If you would like to define serial port 0,
you cannot connect the display to the WiFi connector, but you would have to use the TFT connector on the SKR

**Comment by andyrok** [comment]
ok, but the line #define DGUS_LCD_UI_MKS is not present
<img width="951" alt="Schermata 2022-09-11 alle 19 07 37" src="https://user-images.githubusercontent.com/49729028/189540141-b84e99c3-d36b-48ea-b80e-28fee42b4545.png">

**Comment by radek8** [comment]
![image](https://user-images.githubusercontent.com/46979052/189543496-e6db63f2-af5d-46c3-90da-13938717c820.png)

**Comment by andyrok** [comment]
Ok, so I have to compile Marlin ... not using the specific firmware of the SKR 1.4 Turbo...?
Dwin set folder I already installed it in the H43.... I put the microsd and the Tft update was successful...
Your help is very important ... thank you!!
